snc: generate ca cert for snc if not provided by user

anjannath opened the pull request:
for the openshift-snc service user needs to supply a self signed cert for the admin kubeconfig, this commit adds code to generate this cert when the user has not provided one during create using the flag '--ca-cert-file'

adrianriobo commented on cmd/mapt/cmd/aws/services/openshift-snc.go:
```diff
@@ -72,9 +72,10 @@ func createSNC() *cobra.Command {
-					CaCertFile:     viper.GetString(caCertFile),
-					Spot:           viper.IsSet(awsParams.Spot),
-					Timeout:        viper.GetString(params.Timeout)}); err != nil {
+					CaCertFile: util.If(viper.GetString(caCertFile) == "",
```
mmm I would suggest to remove default value for caCert param...and use `viper.IsSet` to add the value if exists as param....also in case it is not a param..you can leave it as nil...and the handle the creation inside of the action...not rom here...as all logic should be at action level

adrianriobo commented:
may it is worthwhile  if the cert is generated by mapt ...to add it as an output and write it to a file...in case the user woould need it to set something on his env.

Otherwise this LGTM and if we want to write we can do it on a different PR

adrianriobo reviewed (approved):
LGTM

I still see WIP so up to you check comments or just merge

anjannath commented:
> may it is worthwhile if the cert is generated by mapt ...to add it as an output and write it to a file...in case the user woould need it to set something on his env.
> 
> Otherwise this LGTM and if we want to write we can do it on a different PR

yes, currently it writes to the working dir, i'll update to write it in the outputs dir and handle it as a pulumi output

adrianriobo commented:
Ohh I see...but yeah may write it as other outputs so we can ensure all goes to same path based on the output path param

anjannath commented:
> Ohh I see...but yeah may write it as other outputs so we can ensure all goes to same path based on the output path param

there are two scenarios, one is where the user provides the cert in that case we don't want to again write/copy it to the output path, but in the situation where `mapt` has generated the cert it'll be good to write it, but its hard to detect from provider/actions code which scenario happened

adrianriobo commented on pkg/provider/aws/action/openshift-snc/openshift-snc.go:
```diff
@@ -276,6 +289,11 @@ func manageResults(stackResult auto.UpResult, prefix *string) error {
 	}
+
+	caCertPath, ok := stackResult.Outputs[caCertKey].Value.(string)
+	if ok {
```
probably worth it...

anjannath commented:
> This LGTM so feel free to merge when you feel it is finished ^^ @anjannath

ack, thanks.. but in https://github.com/crc-org/snc/pull/1088 we are removing the need for the user to provide the CA files, so the `--ca-cert` flag can be removed from the `mapt aws openshift-snc create` command

adrianriobo commented:
I see.... fair enough.... although now with scenario 1 ensure we get the CA from the machine (as we do with Kubeconfig) as potentially they may need st that CA somewhere while using the cluster

adrianriobo commented:
Also if possible link the issue / pr from snc here

anjannath commented:
> I see.... fair enough.... although now with scenario 1 ensure we get the CA from the machine (as we do with Kubeconfig) as potentially they may need st that CA somewhere while using the cluster
> 
> .... we get and export it (CA) as the kubeconfig

yes, the CA is embedded in the `kubeconfig` file we export, so that should be good to access the cluster

anjannath commented:
@adrianriobo Tested the latest iteration with 4.19 snc bundles and this is now ready for another review, I've updated it as per the comments, and added a new commit to copy the kubeconfig file for output after the `ocp-cluster-ca.service` has ran

adrianriobo reviewed (approved):
LGTM
